public/app/percona/shared/components/hooks/catchCancellationError.ts

```diff
@@ -0,0 +1,38 @@
+import { logger } from '@percona/platform-core';
+import { SerializedError } from '@reduxjs/toolkit';
+import { PERCONA_CANCELLED_ERROR_NAME } from '../../core/constants';
+import { isApiCancelError } from '../../helpers/api';
+
+// Used to safely return from a side effect that might trigger when the component unmounted
+// and some promise resolved in the meanwhile
+export const useCatchCancellationError = () => {
+  const catchFromAsyncThunkAction = async <T>(p: Promise<T>): Promise<T | void> => {
+    const data = await p;
+    //@ts-ignore
```

Comment:
Why is it here?

Reply:
Yeah, typescript wouldn't recognize the error since I'm using a generic type. Just used `&` to allow that